biosnoop broken and dealing with multiple old versions

```
# ./biosnoop.py 
/virtual/main.c:80:26: error: use of undeclared identifier 'REQ_WRITE'; did you mean 'REQ_OP_WRITE'?
    if (req->cmd_flags & REQ_WRITE) {
                         ^~~~~~~~~
                         REQ_OP_WRITE
include/linux/blk_types.h:233:2: note: 'REQ_OP_WRITE' declared here
        REQ_OP_WRITE,
        ^
1 warning and 1 error generated.
Traceback (most recent call last):
  File "./biosnoop.py", line 112, in <module>
    """, debug=0)
  File "/usr/lib/python2.7/dist-packages/bcc/__init__.py", line 166, in __init__
    raise Exception("Failed to compile BPF module %s" % src_file)
Exception: Failed to compile BPF module 
```

Latest linux changed REQ_WRITE to REQ_OP_WRITE.

Easy fix, but it'll then break for 4.4 users. We could have the script detect both, but that way may lie madness: having scripts with kernel version stuff baked in.

I'm inclined to keep /tools just for the latest kernel, and /tools/old for older versions.

But /tools/old already has an older biosnoop.py! So... Make a series? biosnoop.0.py, biosnoop.1.py?

--- a/tools/biosnoop.py
+++ b/tools/biosnoop.py
@@ -98,11 +98,23 @@ int trace_req_completion(struct pt_regs *ctx, struct request *req)
                        req->rq_disk->disk_name);
     }
 
-    if (req->cmd_flags & REQ_WRITE) {
+/*
+ * The following deals with a kernel version change (in mainline 4.7, although
+ * it may be backported to earlier kernels) with how block request write flags
+ * are tested. We handle both pre- and post-change versions here. Please avoid
+ * kernel version tests like this as much as possible: they inflate the code,
+ * test, and maintenance burden.
+ */
+#ifdef REQ_WRITE
+if (req->cmd_flags & REQ_WRITE) {
+#else
+if ((req->cmd_flags >> REQ_OP_SHIFT) == REQ_OP_WRITE) {
+#endif
         data.rwflag = 1;
     } else {
         data.rwflag = 0;
     }
+
     events.perf_submit(ctx, &data, sizeof(data));
     start.delete(&req);
     infobyreq.delete(&req);

